bytecomp/bytegen.ml

```diff
@@ -560,7 +560,7 @@ let rec comp_expr env exp sz cont =
         end
       end
   | Lsend(kind, met, obj, args, _) ->
-      let args = if kind = Cached then List.tl args else args in
+      assert (kind <> Cached);
```

Comment:
Is it the usual practice to do `assert <bool>` to check invariants? I was thinking it would be more informative to fail with a message, e.g.

```ocaml
| Lsend(Cached, _, _, _, _) ->
  invalid_arg "Bytecode does not support Lsend (Cached, ...)"
| Lsend(kind, met, obj, args, _) ->
  ...
```

Reply:
The advantage of an assertion is that the error message gives you the location in the source code, nice for an internal compiler error. We used to have fatal errors named "ctype(1)", "ctype(2)".
But it could be nicer to have also a message.